sdk/core/azure-core/azure/core/_pipeline_client_async.py

```diff
@@ -63,6 +63,26 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+class _AsyncContextManager(collections.abc.Awaitable):
+
+    def __init__(self, wrapped: collections.abc.Awaitable):
+        super().__init__()
+        self.wrapped = wrapped
+        self.response = None
+
+    def __await__(self):
+        return self.wrapped.__await__()
+
+    async def __aenter__(self):
+        self.response = await self
+        return self.response
+
+    async def __aexit__(self, *args):
+        await self.response.__aexit__(*args)
+
+    async def close(self):
+        await self.response.close()
```

Comment:
self.response could be a pipeline response, right?

Reply:
no, this is an httpresponse

Reply:
It is the return of _make_pipeline_call() right?

It looks to me we can return pipeline response by setting "_return_pipeline_response"

Reply:
yeah that's a good point, I've filed an issue to fix it in a separate pr, it's not quite related to this PR, and since `_return_pieline_response` is private, I think we're ok https://github.com/Azure/azure-sdk-for-python/issues/20480